src/main/studio/kdb/QErrors.java

```diff
@@ -12,38 +12,49 @@ public static String lookup(String s) {
         map.put("length","incompatible lengths, e.g. 1 2 3 4 + 1 2 3");
-        map.put("limit","tried to generate a list longer than 2,000,000,000");
+        map.put("limit","trying to generate a list longer than 2^40-1, or serialized object is > 1TB, or too many constants in a function");
         map.put("loop","dependency loop");
         map.put("mismatch","columns that can't be aligned for R,R or K,K ");
-        map.put("Mlim","more than 999 nested columns in splayed tables");
+        map.put("Mlim","more than 65530 nested columns in splayed tables");
+        map.put("noamend","can't change global state inside an amend");
+        map.put("nosocket","trying use sockets in a thread other than the main thread");
+        map.put("noupdate","trying to update state while blocked with -b cmd line arg or reval, or in a thread other than the main thread");
         map.put("nyi","not yet implemented - suggests the\noperation you are tying to do makes sense\nbut it has not yet been implemented");
         map.put("os","operating system error");
-        map.put("pl","peach can't handle parallel lambda's (2.3 only)");
+        map.put("par","unsupported operation on a partitioned table");
+        map.put("parse","invalid syntax, bad IPC header or bad binary data in file");
+        //map.put("pl","peach can't handle parallel lambda's (2.3 only)"); //obsolete
         map.put("Q7","nyi op on file nested array");
-        map.put("rank","invalid rank or valence");
+        map.put("rank","invalid rank");
         map.put("splay","nyi op on splayed table");
         map.put("stack","ran out of stack space");
-        map.put("stop","user interrupt(ctrl-c) or time limit (-T)");
+        map.put("stop","user interrupt (ctrl-c) or time limit (-T)");
         map.put("stype","invalid type used to signal");
-        map.put("type","wrong type, e.g `a+1");
+        map.put("type","wrong type, e.g `a+1, or trying to serialize a nested object which has > 2 billion elements");
         map.put("value","no value");
         map.put("vd1","attempted multithread update");
-        map.put("wsfull","malloc failed. ran out of swap (or addressability on 32bit). or hit -w limit.");
-        map.put("branch","a branch(if;do;while;$[.;.;.]) more than 255 byte codes away");
+        //map.put("wsfull","malloc failed. ran out of swap (or addressability on 32bit). or hit -w limit."); //this is an untrappable error
+
+        map.put("branch","a branch(if;do;while;$[.;.;.]) too many byte codes away"); //the exact number is useless info to an end user
         map.put("char","invalid character");
-        map.put("constants","too many constants (max 96)");
-        map.put("globals","too many global variables (32 max)");
-        map.put("locals","too many local variables (24 max)");
-        map.put("params","too many parameters (8 max)");
+        //map.put("constants","too many constants (max 96)"); //obsolete in 3.6
```

Comment:
Can't hurt to leave it in.  Someone out there is still running 2.8...

Reply:
See my comment on the issue regarding versions.